src/parser/ad_parser.js

```diff
@@ -101,29 +112,42 @@ function parseAdElement(adTypeElement, emit) {
                     creativeTypeElement,
                     creativeAttributes
                   );
-                  if (parsedCreative) {
-                    ad.creatives.push(parsedCreative);
-                  }
                   break;
                 case 'NonLinearAds':
                   parsedCreative = parseCreativeNonLinear(
                     creativeTypeElement,
                     creativeAttributes
                   );
-                  if (parsedCreative) {
-                    ad.creatives.push(parsedCreative);
-                  }
                   break;
                 case 'CompanionAds':
                   parsedCreative = parseCreativeCompanion(
                     creativeTypeElement,
                     creativeAttributes
                   );
-                  if (parsedCreative) {
-                    ad.creatives.push(parsedCreative);
-                  }
                   break;
               }
+
+              if (parsedCreative) {
+                if (universalAdIdElement) {
+                  parsedCreative.universalAdId = {
+                    idRegistry: universalAdIdElement.getAttribute('idRegistry'),
+                    value: parserUtils.parseNodeText(universalAdIdElement)
+                  };
```

Comment:
maybe we could set this as a constant already at the top before the for loop, and just assign it to `parsedCreative.universalAdId` when there is `parsedCreative`. This is just so we are not calling these functions on every loop. Same for creative extensions

Reply:
That wont work, since we want to associate the universalAdId with the actual Creative Element. If I move it outside the for, it basically gets associated to another element, since the array creativeElement.childNodes has a bunch of other elements when parsed.

Reply:
I think we misunderstood each other, I meant that this could be moved to line 32, like:
```
// right after defining universalAdIdElement
const universalAdId = {
    idRegistry: universalAdIdElement.getAttribute('idRegistry') || 'unknown',
    value: parserUtils.parseNodeText(universalAdIdElement)
}

// ...later inside the for loop

parsedCreative.universalAdId = universalAdId
```